src/core/build_env.go

```diff
@@ -54,8 +54,10 @@ func BuildEnvironment(state *BuildState, target *BuildTarget, test bool) []strin
 	env := GeneralBuildEnvironment(state.Config)
 	env = append(env, "PKG="+target.Label.PackageName, "PKG_DIR="+target.Label.PackageDir())
 	if !test {
+		tmpDir := path.Join(RepoRoot, target.TmpDir())
 		env = append(env,
-			"TMP_DIR="+path.Join(RepoRoot, target.TmpDir()),
+			"TMP_DIR="+tmpDir,
+			"TMPDIR="+tmpDir,
```

Comment:
TMPDIR is the POSIX version that things honour for making temp files etc. Since we're not running in a container filesystem any more they need to not all use /tmp otherwise tests can fight with one another.

Indeed the ones that did fail were mostly Java, but that's just coincidence really.